### PR TITLE
Update for Submission

### PR DESCRIPTION
**Pull request by gwaygenomics**
Updating plots for publication. Saving to `.pdf` and replacing `distplot()` with `jointplot()`. I switch string formatting file names and all lowercase variables. Lastly, I add a script that visualizes GWAS signals across TAD boundaries.

Also updating final scripts for reproducing BMD analysis.

### CODE REVIEW

File: scripts/visualize_gc_and_divergence.py [hunk 1]
```diff
@@ -40,33 +41,37 @@
...
+base_file = '{}_{}'.format(genome, tad_cell)
+repeat_index = os.path.join('index',
+                            'REPEATS_index_{}.tsv.bz2'.format(base_file))
+gc_fig_file = os.path.join('figures', genome,
+                           'gc_distribution_{}.pdf'.format(base_file))
+div_fig_file = os.path.join('figures', genome,
+                            'repeat_divergence_{}.pdf'.format(base_file))
+alu_fig_file = os.path.join('figures', genome,
+                            'alu_divergence_{}.pdf'.format(base_file))
+tad_loc = os.path.join(base_dir, '{}_domains_{}.bed'.format(tad_cell, genome))
```
Comment (amyecampbell) [inline scripts/visualize_gc_and_divergence.py]: nice formatted strings

File: scripts/assign_evidence_to_TADs.py [hunk 2]
```diff
@@ -84,65 +79,43 @@ def parse_ev_key(tadkey):
...
-        # Build the key to lookup TAD in dict and lookup
-        chrom, TADkey = buildTADkey(tad_sub)
-        FullTADinfo = TADdictgenes[chrom][TADkey]
+for tad_row in range(len(tad_gwas_df)):
+    snp_info = tad_gwas_df.ix[tad_row, :]
 
-        # Build evidence key
-        e_key = 'chr{}:{}'.format(chrom, TADkey)
+    genes = str(snp_info['gene']).split(',')
+    genes = [x.strip(' ') for x in genes]
```
Comment (dhimmel) [inline scripts/assign_evidence_to_TADs.py]: You don't want to just strip all whitespace (default)?

Comment (gwaygenomics) [inline scripts/assign_evidence_to_TADs.py]: this ended up being unnecessary anyway - should have been deleted in update

File: scripts/visualize_gwas_distribution.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+        # Assign TAD information
+        elem_sub_df['TAD_id'] = tad_id
+        elem_sub_df['TAD_start'] = start
+        elem_sub_df['TAD_end'] = end
+
+        big_out_df = big_out_df.append(elem_sub_df, ignore_index=True)
+
+    return big_out_df
+
```
Comment (amyecampbell) [inline scripts/visualize_gwas_distribution.py]: Is there a reason you set the plot dimensions before defining this function?

File: scripts/visualize_gwas_distribution.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+# Load Constants
+num_bins = 50
+xlab = [''] * num_bins
```
Comment (amyecampbell) [inline scripts/visualize_gwas_distribution.py]: is num_bins variable? If not, is this [' '] * num_bins for readability? (as opposed to just [' '] * 50)

File: scripts/assign_evidence_to_TADs.py [hunk 5]
```diff
@@ -84,65 +79,43 @@ def parse_ev_key(tadkey):
...
-        loop_idx = 0
+        ID, chrom, start, end, ucsc = parse_ev_key(tadkey)
         for evidence in evidence_dict[tadkey][0]:
-            gene, e = evidence
-            if loop_idx == 0:
-                tadwriter.writerow((ID, chrom, start, end, ucsc, gene, e))
-            else:
-                tadwriter.writerow([None] * 5 + [gene, e])
-            loop_idx += 1
+            gene, evidence = evidence
```
Comment (amyecampbell) [inline scripts/assign_evidence_to_TADs.py]: This bit's a little confusing because of the 'evidence' var name

File: scripts/visualize_gwas_distribution.py (hunk 4 above)
Comment (gwaygenomics) [inline scripts/visualize_gwas_distribution.py]: yeah! `num_bins` is used throughout the script. For example, see line 87.

File: scripts/visualize_gwas_distribution.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+for x in range(0, 50, 10):
```
Comment (cgreene) [inline scripts/visualize_gwas_distribution.py]: Is this related to the constant above?

File: scripts/visualize_gwas_distribution.py (hunk 4 above)
Comment (cgreene) [inline scripts/visualize_gwas_distribution.py]: put in all caps and at the top of the script?

File: scripts/visualize_gwas_distribution.py (hunk 3 above)
Comment (gwaygenomics) [inline scripts/visualize_gwas_distribution.py]: yeah, it looks like the default plot arguments should be placed after the function - I'll update this in a new pull request since this one is merged already!

File: scripts/assign_evidence_to_TADs.py (hunk 5 above)
Comment (gwaygenomics) [inline scripts/assign_evidence_to_TADs.py]: yeah, it is pretty confusing...I'll update the variable name in a new pull request!
